### ISSUE TITLE
Production performance is worse than in development

### ISSUE THREAD

**Issue by sanohin** [cerebral/overmind#209]
Hi, thank you for your work. I really liked the library.
I wanted to add a benchmark with overmind to https://github.com/krausest/js-framework-benchmark, but I noticed that in development performance is fine (it took ~200ms to add 1000 items) while in production it is about 4-5 times slower. Can you explain why it is slower and do you have any plans to improve production perf?

**Comment by christianalfoni** [comment]
@sanohin Hi there and thanks a bunch for helping out and testing. This is indeed extremely weird. The library does a lot less work in production so there must be something specifically related to this scenario or we have a serious bug that has just not appeared until now.

Do you happen to have a fork ready with Overmind?

**Comment by christianalfoni** [comment]
Okay, I tried to replicate the issue. 

I refactored the `react-mobx` example and ran both development and production. I also got about 200ms on adding 1000 rows.

Not sure how you got that 4-5 times slower version? But please share and we can look closer :)

Just a note on these benchmarks is that they run optimal code. But the point with libraries like Mobx and Overmind is that you are not able to write optimal code on big apps. Typically with Redux you are exposing too much state to a component and it renders a lot more than it really should. This does not happen with other solutions. So even though optimal code is slower with these kinds of solutions, the overall performance of your app is better because you do not have to take responsibility of how you expose state related to how often the component renders.

But you putting up this issue is really great! It can pin point bugs, so super valuable, thanks a bunch :)

**Comment by sanohin** [comment]
Here is my fork: https://github.com/sanohin/js-framework-benchmark/tree/react-overmind
I copy-pasted react-easy-state implementation with small changes. Not everything works there.
I have about 200ms on adding 1000 rows and 800ms same operation when I build the app in production. I understand that comparing optimal implementation in overmind in redux is pointless, but comparing with libraries that are use Proxy/getters seems to be reasonable

**Comment by christianalfoni** [comment]
Great, thanks for sharing! I will have a look :)

**Comment by sanohin** [comment]
btw, you say it is still a release candidate on the home page. Do you have any roadmap? When do you plan to release a stable version or maybe the latest one is stable enough?

**Comment by christianalfoni** [comment]
Hm... could it be that `setTimeout` is somehow manipulated in the benchmark? There is some very strange behaviour related to it?

**Comment by christianalfoni** [comment]
Oh wait, no... I see issue, though have to dive into why it happens

**Comment by christianalfoni** [comment]
Hm... this has to do with the benchmark. When running "npm run build-dev" it is actually building Overmind in production mode. So the exact same code is executing.

```
		new webpack.DefinePlugin({
			'process.env.NODE_ENV': '"production"'
		})
```

Hm, very weird stuff

**Comment by christianalfoni** [comment]
Or it could be React behaving differently, let me check that

**Comment by christianalfoni** [comment]
Yeah, hm... been looking into this and I can not see what causes it. But it is the production build running both times. If you remove the DefinePlugin you will see that it takes around 500ms to create the rows and you add it back it takes around 150-200ms (which is the production version). 

Please look more into it if you have time and motivation, though I have a few other things I need to focus on and consider this for now.... weird, but not an issue in actual production :)
